### PR TITLE
Enable TF32 for CUDA

### PR DESCRIPTION
**Pull request by mariecwhite**

### CODE REVIEW

File: build_tools/benchmarks/shark/run_shark.sh [hunk 1]
```diff
@@ -33,11 +33,24 @@ git reset --hard ${SHARK_SHA}
...
+  --benchmark
+  --update_tank
+  --maxfail=500
+  -k "${BENCHMARK_REGEX}"
+)
+
+if [[ ${DRIVER} == "cuda" ]]; then
+  args+=(
+    --tf32
+  )
```
Comment (pzread) [inline build_tools/benchmarks/shark/run_shark.sh:46]: nit: Could be written in a single line

File: .github/workflows/run_shark_tank.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
   schedule:
     - cron: '0 10 * * *'
   workflow_dispatch:
+  pull_request:
```
Comment (pzread) [inline .github/workflows/run_shark_tank.yml:7]: Reminder of removing this before submit.

File: build_tools/benchmarks/shark/run_shark.sh (hunk 1 above)
Comment (mariecwhite) [reply build_tools/benchmarks/shark/run_shark.sh]: Done

**Review by GMNGeoffrey — APPROVED** [review]

File: .github/workflows/run_shark_tank.yml (hunk 2 above)
Comment (mariecwhite) [reply .github/workflows/run_shark_tank.yml]: Removed

**Pull request merged**